worlds/kh1/Rules.py

```diff
@@ -1,41 +1,69 @@
-TORN_PAGES = ["Torn Page " + str(i) for i in range(1,6)]
-WORLDS =    ["Wonderland", "Olympus Coliseum", "Deep Jungle", "Agrabah",      "Monstro",      "Atlantica", "Halloween Town", "Neverland",  "Hollow Bastion", "End of the World"]
-KEYBLADES = ["Lady Luck",  "Olympia",          "Jungle King", "Three Wishes", "Wishing Star", "Crabclaw",  "Pumpkinhead",    "Fairy Harp", "Divine Rose",    "Oblivion"]
+from .Locations import KH1Location, location_table
+from .Items import KH1Item, item_table, get_items_by_type, get_items_by_category
 
-def has_x_worlds(state: CollectionState, player: int, num_of_worlds: int, keyblades_unlock_chests: bool) -> bool:
-    worlds_acquired = 0.0
-    for i in range(len(WORLDS)):
-        if state.has(WORLDS[i], player):
-            worlds_acquired = worlds_acquired + 0.5
-        if (state.has(WORLDS[i], player) and (not keyblades_unlock_chests or state.has(KEYBLADES[i], player))) or (state.has(WORLDS[i], player) and WORLDS[i] == "Atlantica"):
-            worlds_acquired = worlds_acquired + 0.5
-    return worlds_acquired >= num_of_worlds
+WORLDS =    ["Destiny Islands", "Traverse Town", "Wonderland", "Olympus Coliseum", "Deep Jungle", "Agrabah",      "Monstro",      "Atlantica", "Halloween Town", "Neverland",  "Hollow Bastion", "End of the World", "100 Acre Wood"]
+KEYBLADES = ["Oathkeeper",      "Lionheart",     "Lady Luck",  "Olympia",          "Jungle King", "Three Wishes", "Wishing Star", "Crabclaw",  "Pumpkinhead",    "Fairy Harp", "Divine Rose",    "Oblivion",         "Spellbinder"]
+BROKEN_KEYBLADE_LOCKING_LOCATIONS = [
+    "End of the World Final Dimension 2nd Chest",
+    "End of the World Final Dimension 4th Chest",
+    "End of the World Final Dimension 7th Chest",
+    "End of the World Final Dimension 8th Chest",
+    "End of the World Final Dimension 10th Chest",
+    "Neverland Hold Aero Chest",
+    "Hollow Bastion Library 1st Floor Turn the Carousel Chest", 
+    "Hollow Bastion Library Top of Bookshelf Turn the Carousel Chest",
+    "Hollow Bastion Library 2nd Floor Turn the Carousel 1st Chest",
+    "Hollow Bastion Library 2nd Floor Turn the Carousel 2nd Chest",
+    "Hollow Bastion Entrance Hall Emblem Piece (Chest)",
+    "Atlantica Sunken Ship In Flipped Boat Chest",
+    "Atlantica Sunken Ship Seabed Chest",
+    "Atlantica Sunken Ship Inside Ship Chest",
+    "Atlantica Ariel's Grotto High Chest",
+    "Atlantica Ariel's Grotto Middle Chest",
+    "Atlantica Ariel's Grotto Low Chest",
+    "Atlantica Ursula's Lair Use Fire on Urchin Chest",
+    "Atlantica Undersea Gorge Jammed by Ariel's Grotto Chest",
+    "Atlantica Triton's Palace White Trinity Chest",
+    "Atlantica Sunken Ship Crystal Trident Event"
+]
 
-def has_emblems(state: CollectionState, player: int, keyblades_unlock_chests: bool) -> bool:
+def has_x_worlds(state: CollectionState, player: int, num_of_worlds: int, keyblades_unlock_chests: bool, logic_difficulty: int) -> bool:
+    if logic_difficulty > 14:
```

Comment:
```suggestion
    if logic_difficulty >= "minimal":
```
As of 0.6.3, you can use inequality operators on string option names for choice options.
If not this, then at least would recommend doing `if logic_difficulty == "minimal":`
Mainly, it's good to avoid magic numbers since I had to go look up what 14 meant here

Reply:
I'm getting `TypeError: '>=' not supported between instances of 'int' and 'str'`.  Is there something special I need to do here?  I can change them to enums if we aren't sure.

Reply:
Is your branch up to date with main?

Reply:
It is unless I'm losing my sanity :p

Reply:
Oh right, this is just an int instead of the option class.
Would recommend passing in the option class instead of just an int then so that you can compare to something that isn't a magic number.

Reply:
https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/witness/hints.py#L132